crates/bevy_render/src/render_graph/node.rs

```diff
@@ -6,28 +6,12 @@ use crate::{
     renderer::RenderContext,
 };
 use bevy_ecs::world::World;
-use bevy_utils::Uuid;
+use bevy_render_macros::define_atomic_id;
 use downcast_rs::{impl_downcast, Downcast};
 use std::{borrow::Cow, fmt::Debug};
 use thiserror::Error;
 
-/// A [`Node`] identifier.
-/// It automatically generates its own random uuid.
-///
-/// This id is used to reference the node internally (edges, etc).
-#[derive(Copy, Clone, Debug, Eq, PartialEq, Ord, PartialOrd, Hash)]
-pub struct NodeId(Uuid);
```

Comment:
NodeId doesn't even need to be globally unique. We could get away with a local non-atomic counter.

Reply:
Hmm true, but this should be part of a separate PR. We should probably create an issue for that.